lib/semantics/check-do.cc

```diff
@@ -139,18 +139,13 @@ class DoConcurrentBodyEnforce {
   void Post(const parser::UnlockStmt &) { NoImageControl(); }
   void Post(const parser::StopStmt &) { NoImageControl(); }
 
-  void Post(const parser::AllocateStmt &) {
-    if (anyObjectIsCoarray()) {
```

Comment:
`anyObjectIsCoarray()` was a place holder which was used in other places as well. Maybe all occurrences of this function should be replaced with whatever logic you inserted to check for Coarrays.

Reply:
It is used on line 175 of `check-do.cc` 
https://github.com/flang-compiler/f18/blob/fc53f3415ee67bf731287e44593fa51ef4fd82ec/lib/semantics/check-do.cc#L175

Reply:
You're probably correct.  I'm working my way through all of the remaining semantic checks for DO constructs, and this change seemed pretty well contained, so I decided to submit a pull request now rather than adding more stuff.